cogs/get_token_authorisation.py

```diff
@@ -19,90 +20,93 @@
-                "does not have admin access to any societies."
+                "does not have admin access to any societies.",
             )
-            logger.debug(NO_ADMIN_DEBUG)
             await ctx.respond("The user token provided does not have any admin access!")
             return
 
-        orgs = [li.get_text(strip=True) for li in parsed_html.find_all("li")]
-
-        DIV_ID: Final[str] = "profile_main"
-        profile_section: bs4.Tag | bs4.NavigableString | None = BeautifulSoup(
+        profile_section_html: bs4.Tag | bs4.NavigableString | None = BeautifulSoup(
             response_html,
             "html.parser",
-        ).find(
-            "div",
-            {"id": DIV_ID},
-        )
+        ).find("div", {"id": "profile_main"})
 
-        if profile_section is None:
-            logger.warning("Couldn't find the profile of the user!")
-            logger.debug(response_html)
+        if profile_section_html is None:
+            logger.warning(
+                "Couldn't find the profile section of the user"
+                "when scraping the website's HTML!",
+            )
+            logger.debug("Retrieved HTML: %s", response_html)
             await ctx.respond(
                 "Couldn't find the profile of the user!"
                 "This should never happen, please check the logs!",
             )
             return
 
-        user_name: bs4.Tag | bs4.NavigableString | int | None = profile_section.find("h1")
+        user_name: bs4.Tag | bs4.NavigableString | int | None = profile_section_html.find("h1")
 
-        if type(user_name) is not bs4.Tag:
-            logger.debug("Found user profile but couldn't find their name!")
-            await ctx.respond("Found user profile but couldn't find the name!")
-            return
+        NO_PROFILE_DEBUG_MESSAGE: Final[str] = "Found user profile but couldn't find their name!"  # noqa: E501
 
-        user_name_str: str = user_name.text
-
-        token_message: str = f"Admin token has access to the following MSL Organisations as {user_name_str}:\n{', \n'.join(org for org in orgs)}"  # noqa: E501
-
-        logger.debug("Admin Token has admin access to: %s as user %s", orgs, user_name_str)
-        await ctx.respond(token_message)
+        if not isinstance(user_name, bs4.Tag):
+            logger.debug(NO_PROFILE_DEBUG_MESSAGE)
+            await ctx.respond(NO_PROFILE_DEBUG_MESSAGE)
+            return
 
+        organisations = [
+            list_item.get_text(strip=True)
+            for list_item
+            in parsed_html.find_all("li")
+        ]
 
+        logger.debug(
+            "Admin Token has admin access to: %s as user %s",
+            organisations,
+            user_name.text,
+        )
 
+        await ctx.respond(
+            f"Admin token has access to the following MSL Organisations as "
+            f"{user_name.text}:\n{', \n'.join(
+                organisation for organisation in organisations
+            )}",
+        )
```

Comment:
Missing `ephemeral=True` argument.

Reply:
that was intentional but happy to defer if you think there's a good reason it should be ephemeral instead

Reply:
If a committee member runs this command in a non-committee channel it will be leaked for all other users to view the output.

Perhaps a better way is to determine whether the "Guest" role can view the channel the response will be sent in, and if they can then set ephemeral to True:

```python
guest_role: discord.Role | None = None
with contextlib.suppress(GuestRoleDoesNotExistError):
    guest_role = bot.guest_role  # Get the bot correctly here

await ctx.respond(
    ...,
    ephemeral=guest_role and ctx.channel.permissions_for(guest_role).is_superset(
        discord.Permissions(send_messages=True, view_channel=True),
    ),
)
```

Reply:
I wonder if it might also be more useful to introduce a command check to see if the command is being run in a committee channel?

Reply:
I think it's still beneficial to allow the command to be run in DMs, rather than limiting it to being a guild only command